iothub/device/src/Authentication/AuthenticationMethodFactory.cs

```diff
@@ -42,7 +42,7 @@ internal static IAuthenticationMethod GetAuthenticationMethodFromConnectionStrin
                         iotHubConnectionString.SharedAccessSignature);
             }
 
-            throw new InvalidOperationException($"Unsupported authentication method in '{iotHubConnectionString}'.");
+            throw new ArgumentException($"Unsupported authentication method in '{iotHubConnectionString}'.");
```

Comment:
We're throwing an ArgumentException for all other cases where our input did not match the expected value, so it made sense to update this one to fit too. 
Let me know what you think.

Reply:
But we're outside the path of a customer sending this argument, right?

What value do you use for the `paramName` parameter that would make sense to the user?

Reply:
I'm leaning towards not throwing this exception in this method, but instead delegating the check to IotHubConnectionCredentials.Validate() method. This method will call out the specific property that is unset, in this case that being:

https://github.com/Azure/azure-iot-sdk-csharp/blob/327e7b5ecfc791e19efaca9792dd32fb401e73e7/iothub/device/src/Authentication/IotHubConnectionCredentials.cs#L325-L335

Reply:
As long as when the user gets the exception, it matches what they see in the API surface they called.

If needed, you can catch this exception, grab the exception message, and create a new one with the parameter name you know about.

Reply:
Updated to:
```c#
throw new ArgumentException($"Should specify either SharedAccessKey or SharedAccessSignature" +
    $" if connection string is used for authenticating the client with IoT hub.");
```